products/jbrowse-web/src/sessionModelFactory.ts

```diff
@@ -272,7 +278,17 @@ export default function sessionModelFactory(pluginManager: PluginManager) {
       },
 
       deleteConnection(configuration: AnyConfigurationModel) {
-        return getParent(self).jbrowse.deleteConnectionConf(configuration)
+        if (self.adminMode) {
+          return getParent(self).jbrowse.deleteConnectionConf(configuration)
+        }
+        const { connectionId } = configuration
+        const idx = self.sessionConnections.findIndex(
+          c => c.connectionId === connectionId,
+        )
+        if (idx === -1) {
+          return undefined
+        }
+        return self.sessionConnections.splice(idx, 1)
```

Comment:
What if it's a session connection, but we are also in admin mode? Seems to me that this code would not delete the connection.

Reply:
sessionConnections would likely not be used in admin mode (any adding or deleting of connections in adminMode would affect the config.json)

your observation is true, but it might be an acceptable limitation, unless we really want to cover all cases